contracts/twap/CumulativeTwap.sol

```diff
@@ -71,6 +77,13 @@ contract CumulativeTwap is BlockContext {
 
         Observation memory latestObservation = observations[currentObservationIndex];
 
+        // Use latestObservation instead, if the latest updated timestamp is equal to latestObservation's timestamp
+        // it's to be consistent with the logic of _update
+        if (latestObservation.timestamp == latestUpdatedTimestamp) {
+            price = latestObservation.price;
+            latestUpdatedTimestamp = latestObservation.timestamp;
+        }
+
```

Comment:
For v3, the same timestamp's prices will only be sent to `CumulativeTwap` once so here won't happen actually.

If we still need this, how about we set a revert funciton here to prevent the misuse of this method? 
`require(latestObservation.timestamp == latestUpdatedTimestamp ? price == latestObservation.price : true, "error");`

Reply:
After the discussion, let's remain the same because revert function might make a tx gets failed if we neglect this part of logic in the future implementation. 

We could be enhance like below:
```
if (latestObservation.timestamp == latestUpdatedTimestamp && price != latestObservation.price) {
    price = latestObservation.price;
}
```